internal/descriptor/descriptor.go

```diff
@@ -21,6 +21,19 @@ import (
 	"oras.land/oras/internal/docker"
 )
 
+// IsManifest checks if a descriptor describes a manifest.
+func IsManifest(desc ocispec.Descriptor) bool {
```

Comment:
This function seems to be copied from `oras-go` with slight modification. Can you add a reference in the comment block?